### ISSUE TITLE
"From Selection" option on lists nodes not working

### ISSUE THREAD

**Issue by dimitripaiva** [JacquesLucke/animation_nodes#1105]
I'm trying to create a list of objects, and while I can put them manually on a list, the option to create a list from selected objects is not working. 

Thanks!

**Comment by Ekkiwnox** [comment]
Hi ! I've tried the same here and did not work too. I don't know if I'm the only one, but seems like the particle system is also not responding :p

**Comment by 3DSinghVFX** [comment]
Hi @OmarSquircleArt I have fixed the "create from selection" problem,
In selection.py file, added the line `sortedSelectionNames = getSelectedObjectNames()` in getSortedSelectedObjectNames() function, and in create_list .py change the line (number 136) to `socket.object = bpy.data.objects[name]`.

**Comment by OmarSquircleArt** [comment]
@3DSinghVFX Have you tried that? I don't think this is enough to fix the issue. As far as I know, the main issue is not in the name/pointer property. The main issue is caused by the fact that the `Always` handler is implemented using a timer that have no runtime context, which means we don't have access to selected objects. I think we discussed this before at some point but never fixed it. Will ask Jacques about it again.

**Comment by Clockmender** [comment]
I sorted this node some time ago as described below, but I don't think this solution ever made it to a release:

----Begin Previous Post----

OK, this works for me, I'll leave you to correct the "Get Selected Node" for your release as you want it.

![Screen Shot 2019-03-18 at 13 57 55](https://user-images.githubusercontent.com/29657804/54531614-9d481f80-497d-11e9-8dcf-e0673b61e912.png)

![Screen Shot 2019-03-18 at 13 58 10](https://user-images.githubusercontent.com/29657804/54531620-a0431000-497d-11e9-89f9-46b201f8dcc1.png)

Code:

```python
import bpy
from bpy.props import *
from .... events import isRendering, propertyChanged
from .... base_types import AnimationNode
from .... utils.selection import getSortedSelectedObjects

class GetSelectedObjectsNode(bpy.types.Node, AnimationNode):
    bl_idname = "an_GetSelectedObjectsNode"
    bl_label = "Get Selected Objects"
    searchTags = ["Get Active Object"]
    bl_width_default = 200

    message: StringProperty()

    def draw(self, layout):
        if self.message is not '':
            layout.label(text = self.label, icon = "INFO")

    def create(self):
        self.newOutput("Object List", "Selected Objects", "selectedObjects")
        self.newOutput("Object", "Active Object", "activeObject")

    def execute(self):
        if isRendering():
            self.message = 'Disabled During Rendering'
            return [], None
        else:
            label = bpy.context.scene.name + ' Selected: '
            sortedList = sorted([ob for ob in bpy.context.view_layer.objects if ob.visible_get() and ob.select_get()], key=lambda x: x.name)

            self.label = label+str(len(sortedList))
            return sortedList, bpy.context.view_layer.objects.active
```

It seems to stand up to scrutiny and does not show un-attached objects, or invisible objects, etc. I may also alter it to show selected bones - that would be helpful to me, but probably me only.

Cheers, Clock.

----End Previous Post----

I also got it to show selected bones, but you need my "bone" socket also.

![Screen Shot 2019-03-18 at 15 30 21](https://user-images.githubusercontent.com/29657804/54537539-91af2580-498a-11e9-9ff6-50ff6ad9da55.png)

![Screen Shot 2019-03-18 at 15 30 49](https://user-images.githubusercontent.com/29657804/54537548-95db4300-498a-11e9-9bcb-5a928b27d9b2.png)

I can upload this version if anyone is interested.

**Comment by 3DSinghVFX** [comment]
@OmarSquircleArt Yes, it is working.
![Peek 2019-05-21 14-30](https://user-images.githubusercontent.com/30294746/58082832-6983b300-7bd5-11e9-986a-f35585e20532.gif)

**Comment by Clockmender** [comment]
My bad, I didn't read this properly, forget what I said, although it does sort the "Selected Objects" node...

**Comment by OmarSquircleArt** [comment]
@3DSinghVFX This doesn't sort the objects based on the selection order, does it?

**Comment by OmarSquircleArt** [comment]
Fixed in 642ef00.
